packages/react/src/components/phoneInput/PhoneInput.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useRef } from 'react';
+import isFunction from 'lodash.isfunction';
+
+import 'hds-core';
+import { InputWrapper } from '../../internal/input-wrapper/InputWrapper';
+import { TextInputProps } from '../textInput';
+import textInputStyles from '../textInput/TextInput.module.css';
+import comboseAriaDescribedBy from '../../utils/comboseAriaDescribedBy';
+
+export type PhoneInputProps = Omit<
+  TextInputProps,
+  'buttonIcon' | 'buttonAriaLabel' | 'onButtonClick' | 'children' | 'label'
+> & {
+  /**
+   * The label for the input
+   */
+  label?: string;
+};
+
+export const PhoneInput = React.forwardRef<HTMLInputElement, PhoneInputProps>(
+  (
+    {
+      'aria-describedby': customAriaDescribedBy,
+      className = '',
+      disabled = false,
+      defaultValue,
+      errorText,
+      helperText,
+      hideLabel,
+      invalid,
+      id,
+      label,
+      labelText,
+      onChange = () => null,
+      required,
+      style,
+      successText,
+      tooltipLabel,
+      tooltipText,
+      tooltipButtonLabel,
+      type = 'tel',
+      ...rest
+    }: PhoneInputProps,
+    ref?: React.Ref<HTMLInputElement>,
+  ) => {
+    const wrapperProps = {
+      className,
+      errorText,
+      helperText,
+      hideLabel,
+      id,
+      invalid,
+      label,
+      labelText,
+      required,
+      style,
+      successText,
+      tooltipLabel,
+      tooltipText,
+      tooltipButtonLabel,
+    };
+
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    /**
+     * Merge props.ref to the internal ref. This is needed because we need the ref ourself and cannot rely on
+     * component user to provide it.
+     */
+    useEffect(() => {
+      if (ref) {
+        if (isFunction(ref)) {
+          (ref as (instance: HTMLInputElement) => void)(inputRef.current);
+        } else {
+          // eslint-disable-next-line no-param-reassign
+          (ref as React.MutableRefObject<HTMLInputElement>).current = inputRef.current;
+        }
+      }
```

Comment:
I've seen this ref-handling multiple times in our codebase. Would it be possible to make it some util-hook etc.? Or is it a too small code block to abstract?

Reply:
- [ ] Abstract to util function